src/graphic/Fast3D/gfx_dxgi.cpp

```diff
@@ -517,13 +517,11 @@ static void gfx_dxgi_handle_events(void) {
 }
 
 static uint64_t qpc_to_ns(uint64_t qpc) {
-    qpc *= 1000000000;
-    return qpc / dxgi.qpc_freq;
+    return qpc / dxgi.qpc_freq * 1000000000 + qpc % dxgi.qpc_freq * 1000000000 / dxgi.qpc_freq;
```

Comment:
could be nice to make this a little less magic and throw
```cpp
#define MAGIC_NUMBER_NAME 1000000000
```
in there and replace this with

```suggestion
    return qpc / dxgi.qpc_freq * MAGIC_NUMBER_NAME + qpc % dxgi.qpc_freq * MAGIC_NUMBER_NAME / dxgi.qpc_freq;
```

Reply:
Your magic number is just the factor from seconds to ns. Or seconds to 100ns steps respectively.

Reply:
I've updated the PR to add defines for these constants